main.py

```diff
@@ -160,7 +163,21 @@ def bad_request(e):
 
 @app.errorhandler(404)
 def page_not_found(e):
-	return render_template('error/404.html', error=e), 404
+	url = urlparse(request.url)
+	path = url.path
+	if legacy_util.should_redirect(path):
+		page = legacy_util.get_redirect_page(path)
+		redirect_url = url_for(page.name, **page.kwargs)
+		response = make_response(redirect(redirect_url, code=301))
+		# Set a cookie that expires 5 seconds after page load, 
```

Comment:
How confident are you that the page will load in under 5s? ;-)

Reply:
Surprisingly no HTTP Archive data is available in CrUX for me to confirm, but I'm willing to accept the graceful degradation. :smile: